pkg/kratos/handlers.go

```diff
@@ -346,7 +354,9 @@ func (a *API) is40xError(err error) bool {
 	return false
 }
 
-func (a *API) mfaSettingsRedirect(w http.ResponseWriter, returnTo string) {
+func (a *API) mfaSettingsRedirect(w http.ResponseWriter, r *http.Request, returnTo, loginChallenge string) {
```

Comment:
micro-nitpick: if you only need the context from the request, consider just passing the context as an argument instead of the request pointer